formatters/html/html.go

```diff
@@ -563,3 +561,60 @@ func compressStyle(s string) string {
 	}
 	return strings.Join(out, ";")
 }
+
+const styleCacheLimit = 16
+
+type styleCacheEntry struct {
+	style *chroma.Style
+	cache map[chroma.TokenType]string
+}
+
+type styleCache struct {
+	mu sync.Mutex
+	// LRU cache of compiled (and possibly compressed) styles. This is a slice
+	// because the cache size is small, and a slice is sufficiently fast for
+	// small N.
+	cache []styleCacheEntry
+	f     *Formatter
+}
+
+func newStyleCache(f *Formatter) *styleCache {
+	return &styleCache{f: f}
+}
+
+func (l *styleCache) get(style *chroma.Style) map[chroma.TokenType]string {
+	l.mu.Lock()
+	defer l.mu.Unlock()
```

Comment:
Can this be turned into a RWLock? Most cases will probably only write to this cache once so it's be a shame to introduce potential lock-contention if there are many reads.

Something like https://stackoverflow.com/a/70236841 should dodge the gotchas with finishing the read-defer before grabbing the write lock

Reply:
Possibly, send a PR through if you're interested.

Reply:
FWIW I started with an RWMutex but it simplifies the code being a plain Mutex.